pydantic/error_wrappers.py

```diff
@@ -51,8 +51,9 @@ def dict(self, *, loc_prefix: Optional[Tuple[str, ...]] = None) -> Dict[str, Any
 class ValidationError(ValueError):
     __slots__ = ('raw_errors',)
 
-    def __init__(self, errors: Sequence[ErrorList]) -> None:
+    def __init__(self, errors: Sequence[ErrorList], obj_or_type: Any) -> None:
         self.raw_errors = errors
+        self.obj_type: Type[Any] = obj_or_type if isinstance(obj_or_type, type) else type(obj_or_type)
```

Comment:
please add this to slots, also I think `model` or `model_cls` would be a better name.